src/Compilers/CSharp/Portable/Parser/Lexer.cs

```diff
@@ -2237,7 +2240,7 @@ private void LexSyntaxTrivia(bool afterFirstToken, bool isTrailing, ref SyntaxLi
 
                             // normal single line comment
                             this.ScanToEndOfLine();
-                            var text = TextWindow.GetText(false);
+                            var text = TextWindow.GetText(poolTrivia);
```

Comment:
A general approach on what to intern and what not was that "natural language things" are not interned - like comments, xml doc. That is because natuaral language things are not constrained in their content and are not as repetitive as something that follows a formal grammar.

The claim of this change is that the first comment in a file is repetitive since it is typically a copyright header.  
I guess that can be a correct enough assumption for this to work.

My concern would be that we might be interning something that has no size limit. Not even soft/statistical one. In degenerate cases, like whole file is a comment or similar, we may end up retaining huge strings and if user starts editing them, retaining many versions.

That could be mitigated to a degree if we have some size limit (256 chars or smth like that) on interned text - at the level of `GetText` or `Intern`. 
I think we need such safeguard.